packages/callout/src/Callout.spec.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { cleanup, render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Callout, {
+  colorSets,
+  headerIcons,
+  headerLabels,
+  CustomCallout,
+  CustomCalloutProps,
+  Variant,
+} from './Callout';
+
+const className = 'test-classname';
+const children = 'this is the callout content.';
+const title = 'this is the callout title';
+const colorSet = colorSets[Variant.Note];
+const headerIcon = headerIcons[Variant.Note];
+const headerLabel = headerLabels[Variant.Note];
+
+function renderCustomCallout(props: Partial<CustomCalloutProps> = {}) {
+  const defaultProps = {
+    children,
+    colorSet,
+    headerLabel,
+  };
+  return render(<CustomCallout {...defaultProps} {...props} />);
+}
+
+describe('packages/callout', () => {
+  describe('Callout component', () => {
+    test('renders each variant correctly', () => {
+      for (const variant of Object.values(Variant)) {
+        const calloutMarkup = renderToStaticMarkup(
+          <Callout className={className} variant={variant}>
+            {children}
+          </Callout>,
+        );
+
+        const customCalloutMarkup = renderToStaticMarkup(
+          <CustomCallout
+            className={className}
+            colorSet={colorSets[variant]}
+            headerIcon={headerIcons[variant]}
+            headerLabel={headerLabels[variant]}
+          >
+            {children}
+          </CustomCallout>,
+        );
+
+        expect(calloutMarkup).toEqual(customCalloutMarkup);
+      }
+    });
+  });
+
+  describe('CustomCallout component', () => {
+    afterEach(async () => {
+      await cleanup();
+    });
+
+    test(`renders class name in classList: "${className}"`, async () => {
+      let { container } = renderCustomCallout({});
```

Comment:
Recently learned that it's actually preferred to grab this from screen rather than destructuring render. You can import `screen` where you import render from `react-testing-library`: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library